credential/auth.go

```diff
@@ -63,9 +66,18 @@
 		return nil, NewVCError(ErrMalformed, err)
 	}
 
+	proof, err := extractProof(cred)
+	if err != nil {
+		return nil, NewVCError(ErrInvalidProof, err)
+	}
+	if proof.ProofPurpose != ProofPurposeAuthentication {
+		return nil, NewVCError(ErrAuthClaim,
+			fmt.Errorf("proof purpose not targeting `%s` (proof purpose: `%s`)", ProofPurposeAuthentication, proof.ProofPurpose))
+	}
```

Comment:
@coderabbitai I know, I didn't find a test case to handle this error since if there is an error in the proof extraction, error is thrown before when trying parse credential (Ligne 40). If you find a solution to test this error handling at line 71 ?

Reply:
@coderabbitai In practice, it is not possible to encounter this error handling scenario because the `parseSigned` method will throw an error before reaching the error handling on line 71. Additionally, within the `parseSigned` method, `ParseCredential` will attempt to decode and extract the proof, throwing an error before any custom error handling can occur. Therefore, testing this scenario without mocking is not feasible and would require code refactoring.